src/raft.rs

```diff
@@ -971,10 +971,15 @@ impl<T: Storage> Raft<T> {
         true
     }
 
-    /// Notifies that these raft logs have been well persisted.
+    /// Notifies that these raft logs or snapshot have been persisted.
     pub fn on_persist_entries(&mut self, index: u64, term: u64) {
         let update = self.raft_log.maybe_persist(index, term);
-        if update && self.state == StateRole::Leader {
+        // Actually, if it is a leader and persisted index is updated, this term
+        // must be equal to self.term because the persisted index must be equal to
+        // the last index of entries from previous leader when it becomes leader
+        // (see the comments in become_leader), namely, the new persisted entries
+        // must come from this leader. Here checking the term just for robustness.
+        if update && self.state == StateRole::Leader && term == self.term {
```

Comment:
How about using a log? Even if the term doesn't match `self.term` in the future adaption, for example introducing paging, it's still safe to enter the if branch.

Reply:
Done.